wic/src/main/java/edu/asu/diging/wic/core/repository/GraphRepository.java

```diff
@@ -0,0 +1,40 @@
+package edu.asu.diging.wic.core.repository;
+
+import java.util.List;
+
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.diging.wic.core.model.impl.Edge;
+import edu.asu.diging.wic.core.model.impl.Graph;
+
+@Repository
+public interface GraphRepository extends PagingAndSortingRepository<Graph, String> {
+
+    @Query("SELECT g from Graph g WHERE g.conceptUri = :uri")
+    public List<Graph> getGraphs(@Param("uri") String uri);
+
+    public Graph save(Graph graph);
+
+    @Query("SELECT e FROM Edge e WHERE e.sourceNode.uri IN "
+            + "(SELECT alternativeUris FROM ConceptpowerAlternativeUri c WHERE c.id = :uri) "
```

Comment:
have you tried making this:
`SELECT c.alternativeUris FROM tbl_conceptpower_concept  c WHERE c.id = :uri`

The way it is right now does not seem to make sense. `ConceptpowerAlternativeUri` is never used is it? So the table would always be empty? I don't think you need it if you write the query correctly.